ListBuilder/DynamicLabeledListBuilder.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\FormBundle\ListBuilder;
+
+use Sulu\Bundle\FormBundle\Entity\Dynamic;
+use Sulu\Bundle\FormBundle\Entity\Form;
+
+class DynamicLabeledListBuilder extends DynamicListBuilder
+{
+    private $formFieldsLabelCache = [];
+
+    public function build(Dynamic $dynamic, string $locale): array
+    {
+        $entry = parent::build($dynamic, $locale);
+
+        if (empty($entry)) {
+            return $entry;
+        }
+
+        $labels = $this->getLabels($dynamic->getForm(), $locale);
+        $entryLabeled = [];
+
+        foreach ($entry[0] as $key => $value) {
+            $entryLabeled[$labels[$key] ?? $key] = $value;
+        }
+
+        return [$entryLabeled];
+    }
+
+    private function getLabels(Form $form, string $locale): array
+    {
+        if (!isset($this->formFieldsLabelCache[$form->getId()])) {
+            $labels = [];
+
+            foreach ($form->getFields() as $field) {
+                $label = '';
+                $translation = $field->getTranslation($locale, false, true);
+
+                if ($translation) {
+                    $label = $translation->getShortTitle() ?: \strip_tags($translation->getTitle());
```

Comment:
I was testing this and I found an edge case: The "spacer" field does have a translation, but both `getShortTitle` and `getTitle` return `null`- in this case `strip_tags` will fail because it expects a string.

IMHO when there is no translation or all titles are `null`, I would just skip the loop instead of setting an empty string. Then in the `build` method above it will fall back to the key, which is better than nothing.

Reply:
I found two more special fields that need different handling: `freeText` and `headline`.
Both are just there to add additional text to the form, but are no real fields. Before it appeared in the export as "freeText", "freeText1" with no content in the column etc. Now it's using the title which can potentially be a really long text. So either it should be skipped (it has no content anyway) or the key should be used instead.

Reply:
Looks like there is already a constant with these special fields: https://github.com/sulu/SuluFormBundle/blob/2.5/Entity/Dynamic.php#L36

And the `DynamicListBuilder` you are extending has a `getFields` method with a `hideHidden` flag, which does exactly what we want - skip these fields 👍

Reply:
Hi, sorry for the delay. I've updated the PR with your suggestions.